File: Leetcode/twoSum.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
       int index[] = { 0, 0 };

		int len = numbers.length;
		HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();

		for (int i = 0; i < len; i++) {

			map.put(numbers[i], i );

		}
		
		
		for(int i = 0;i < len;i++){
			
			if(map.get(target - numbers[i]) != null && map.get(target - numbers[i]) > i){
				index[0] = i;
				index[1] = map.get(target - numbers[i]);
				
			}
			
			
		
			
		}
		

		return index;
        
    }

public static void main(String agrs[]){
    int[] numbers = {0,4,3,0} ;
    int target = 0;
    int[] result = twoSum(numbers, target);
    System.out.println("index1="+result[0]+", index2="+result[1]);
    //index1=1, index2=2
}



}

File: Leetcode/TwoSum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict_num={}

        result=[0,0]
        for i in range(len(nums)) :
            if((target - nums[i]) in dict_num):
                result[0] = dict_num[target - nums[i]]
                result[1] = i
            else:
                dict_num[nums[i]] = i
            
        
        print(result)
        return result







nums = [3,2,3]
target = 6
test = Solution()
test.twoSum(nums,target)
print("2222")
